cmd/ci-operator/main.go

```diff
@@ -786,6 +789,34 @@ func overrideMultiStageParams(o *options) error {
 	return nil
 }
 
+// overrideENVListParams updates the environment variables in the test configurations based on the OVERRIDE_ENV_LIST configuration.
+// If the configuration exists, it splits the values by commas and adds the corresponding environment variables to each test configuration.
+func overrideENVListParams(o *options) {
+	viper.AutomaticEnv()
+	overrideEnvList := viper.GetString("OVERRIDE_ENV_LIST")
+	if overrideEnvList == "" {
+		return
+	}
+
+	envs := strings.Split(overrideEnvList, ",")
+
+	for _, env := range envs {
+		env = strings.TrimSpace(env)
+		envValue := viper.GetString(env)
+
+		for _, test := range o.configSpec.Tests {
+			if test.MultiStageTestConfigurationLiteral != nil {
+				if test.MultiStageTestConfigurationLiteral.Environment == nil {
+					test.MultiStageTestConfigurationLiteral.Environment = make(api.TestEnvironment)
+				}
+
+				test.MultiStageTestConfigurationLiteral.Environment[env] = envValue
+			}
+		}
+
+	}
+}
```

Comment:
```suggestion
func applyEnvOverrides(o *options) {
	viper.AutomaticEnv()

	for _, envVar := range os.Environ() {
		parts := strings.SplitN(envVar, "=", 2)
		if len(parts) != 2 {
			continue
		}
		key, value := parts[0], parts[1]

		if !strings.HasPrefix(key, "OVERRIDE_") {
			continue
		}

		for _, test := range o.configSpec.Tests {
			if test.MultiStageTestConfigurationLiteral != nil {
				if test.MultiStageTestConfigurationLiteral.Environment == nil {
					test.MultiStageTestConfigurationLiteral.Environment = make(api.TestEnvironment)
				}
				test.MultiStageTestConfigurationLiteral.Environment[key] = value
			}
		}
	}
}
```

Reply:
I would change the logic of this function to pass everything starting with OVERRIDE_ prefix to multistage test.